staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -210,20 +214,21 @@ type SecureServingInfo struct {
 // NewConfig returns a Config struct with the default values
 func NewConfig(codecs serializer.CodecFactory) *Config {
 	return &Config{
-		Serializer:                   codecs,
-		ReadWritePort:                443,
-		RequestContextMapper:         apirequest.NewRequestContextMapper(),
-		BuildHandlerChainFunc:        DefaultBuildHandlerChain,
-		LegacyAPIGroupPrefixes:       sets.NewString(DefaultLegacyAPIPrefix),
-		DisabledPostStartHooks:       sets.NewString(),
-		HealthzChecks:                []healthz.HealthzChecker{healthz.PingHealthz},
-		EnableIndex:                  true,
-		EnableDiscovery:              true,
-		EnableProfiling:              true,
-		MaxRequestsInFlight:          400,
-		MaxMutatingRequestsInFlight:  200,
-		MinRequestTimeout:            1800,
-		EnableAPIResponseCompression: utilfeature.DefaultFeatureGate.Enabled(features.APIResponseCompression),
+		Serializer:                    codecs,
+		ReadWritePort:                 443,
+		RequestContextMapper:          apirequest.NewRequestContextMapper(),
+		BuildHandlerChainFunc:         DefaultBuildHandlerChain,
+		LegacyAPIGroupPrefixes:        sets.NewString(DefaultLegacyAPIPrefix),
+		DisabledPostStartHooks:        sets.NewString(),
+		HealthzChecks:                 []healthz.HealthzChecker{healthz.PingHealthz},
+		EnableIndex:                   true,
+		EnableDiscovery:               true,
+		EnableProfiling:               true,
+		MaxRequestsInFlight:           400,
+		MaxMutatingRequestsInFlight:   200,
+		MinRequestTimeout:             1800,
+		DisconnectChanceOnMaxRequests: 0.01,
```

Comment:
Because it requires no extra configuration for the vast majority of those servers.  I'll turn this around - why would I want a Kubernetes API server that is under load and should / would have multiple peers to continue to accept connections from the server?  I'm certainly not against adding logic for this on the client as well (the client should close idle connections when it receives a 429 possibly).